lib/jobs/copy-publickey.js

```diff
@@ -0,0 +1,66 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+module.exports = copyKeyJobFactory;
+
+di.annotate(copyKeyJobFactory, new di.Provide('Job.CopyKey'));
+di.annotate(copyKeyJobFactory, new di.Inject(
+    'Job.Base',
+    'Util',
+    'Logger',
+    'Assert',
+    'Promise',
+    'Services.Waterline',
+    'ssh',
+    'JobUtils.Commands'
+));
+
+function copyKeyJobFactory(
+    BaseJob,
+    util,
+    Logger,
+    assert,
+    Promise,
+    waterline,
+    ssh,
+    CommandUtil
+) {
+    var logger = Logger.initialize(copyKeyJobFactory);
+    var commandUtil;
+    function CopyKeyJob(options, context, taskId) {
```

Comment:
a little jsdoc noting that context.target should be a nodeID would be potentially useful here